include/engine/phantom_node.hpp

```diff
@@ -127,6 +127,23 @@ struct PhantomNode
 
     bool operator==(const PhantomNode &other) const { return location == other.location; }
 
+    bool IsIndistinct(const PhantomNode &other) const
+    {
+        if (forward_segment_id.enabled && other.forward_segment_id.enabled &&
```

Comment:
why is only Forward *or* Reverse enough?

Reply:
Filtering can decouple forward and reverse edges into different candidates to avoid u-turns. And switch "enable" to false. See [this filter!](https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/plugins/match.cpp#L90-L99)
In the other hand I suppose that different requests to the r-tree index with same point will return same forward and backward results. So I check forward segment id and backward segment id separately.